Source/Core/Core/CoreTiming.cpp

```diff
@@ -305,6 +310,8 @@ void CoreTimingManager::Advance()
     Event evt = std::move(m_event_queue.front());
     std::pop_heap(m_event_queue.begin(), m_event_queue.end(), std::greater<Event>());
     m_event_queue.pop_back();
+
+    Throttle(evt.time, emulation_speed);
```

Comment:
I don't think this is the right place for the `Throttle()` call. The way `CoreTiming` works -- from what I remember, anyway -- is that `CoreTiming::Advance()` is called periodically by the JIT, and by the time it gets control multiple events could have logically already happened but haven't been executed yet (which this loop then executes in order to 'catch up' the event state). So by the time this code runs, we're already at `m_globals.global_timer`, which makes it odd to only throttle until `evt.time`, which may be earlier than current -- and also means you may throttle multiple times in quick succession when once would have sufficed.

It probably makes more sense to put a `Throttle(m_globals.global_timer, emulation_speed)` *after* this loop instead.

This is also most likely the most significant change here, so you should probably mention it in the PR title and description.

Reply:
From what I understand the CPU doesnt actually need to throttle, but throttling here just makes sure that every event happens exactly when it's supposed to. The way the old throttler worked is that it created an event every GetTicksPerSecond() / 1000 that would sleep for about 1ms depending on the timer. This one is a little bit more precise. (having it here is also used in a future PR which uses this throttler to detect lag and skip VI's if needed)